### ISSUE TITLE
TypeDoesNotContainType - false positive string vs. string

### ISSUE THREAD

**Issue by SpacePossum** [vimeo/psalm#2240]
Hi all!

Thanks for providing this great tool. I found a little bug, hope it is good report, let me know if you need more details.

```php
<?php

$repo = new stdclass();

if ('Doctrine\ORM\EntityRepository' === \get_class($repo)) {
    echo 1;
}
```

triggers:
> ERROR: TypeDoesNotContainType - 5:5 - class-string cannot be identical to string(Doctrine\ORM\EntityRepository)

```
$ ./vendor/bin/psalm -v
Psalm 3.2.2@f0ddc6f3bc35ba7ecb99232903907a1d5ec1c8e8
```

https://psalm.dev/r/c2e91b264c

--edit: fix the link

**Comment by muglug** [comment]
Yes, Psalm requires that all classes be expressed as `::class` constants like so: https://psalm.dev/r/d9dc0829d3

**Comment by SpacePossum** [comment]
thanks! sorry I got confused about the description of this check.
